misc/python/materialize/spawn.py

```diff
@@ -117,7 +117,7 @@ def capture(
         whitespace.
     """
     input = None
-    if isinstance(stdin, bytes):
+    if isinstance(stdin, bytes) or isinstance(stdin, str):
```

Comment:
The problem is that `bytes` just doesn't work period. Because `text=True` below, if `input` is provided, it has to be a `string` rather than bytes. (Providing an `IO[bytes]`, however, seems to be fine; go figure.) So I think the correct thing to do is to change this to _just_ check for `str`. Note that you'll also need to adjust the type signature of the method to swap `bytes` for `str`.